### PR TITLE
Missing moderate PEM YLDs bugfix

### PR DESCRIPTION
**Pull request by hussain-jafari**
### Description
- *Category*: bugfix, observers
- *JIRA issue*: [MIC-3361](https://jira.ihme.washington.edu/browse/MIC-3361) 

Explicitly include moderate PEM as cause component in disability observer even if disability weights are all 0. 

### Verification and Testing
Ran single run that was previously not producing YLDs and verified that it was producing desired moderate PEM YLDs.

### CODE REVIEW

File: src/vivarium_gates_child_iv_iron/components/observers.py [hunk 1]
```diff
@@ -27,42 +28,42 @@ def register_stratifications(self, builder: Builder) -> None:
...
-        #     categories={"uncovered", "covered"},
-        # )
-        #
-        # self.setup_stratification(
-        #     builder,
-        #     name="bmi_anemia",
-        #     sources=[Source("maternal_bmi_anemia_exposure", SourceType.COLUMN)],
-        #     categories={"cat4", "cat3", "cat2", "cat1"},
-        # )
+        self.setup_stratification(
```
Comment (rmudambi) [inline src/vivarium_gates_child_iv_iron/components/observers.py:31]: Are these intended to be uncommented now? Were they commented out before just for testing?

File: src/vivarium_gates_child_iv_iron/components/observers.py [hunk 2]
```diff
@@ -79,6 +80,15 @@ def child_growth_risk_factor_stratification_mapper(self, row: pd.Series) -> str:
         }[row.squeeze()]
 
 
+class DisabilityObserver(DisabilityObserver_):
+    def on_post_setup(self, event: Event) -> None:
+        for cause in self._cause_components:
+            if cause.has_disability:
```
Comment (rmudambi) [inline src/vivarium_gates_child_iv_iron/components/observers.py]: better to say `if cause.has_disability or cause.name == 'disease_model...':` in a single if block

File: src/vivarium_gates_child_iv_iron/model_specifications/branches/scenarios.yaml [hunk 3]
```diff
@@ -1,5 +1,5 @@
-input_draw_count: 66
-random_seed_count: 10
+input_draw_count: 25
+random_seed_count: 60
```
Comment (rmudambi) [inline src/vivarium_gates_child_iv_iron/model_specifications/branches/scenarios.yaml:2]: shouldn't this be 120?

**Review by rmudambi — APPROVED** [review]

File: src/vivarium_gates_child_iv_iron/components/observers.py (hunk 1 above)
Comment (hussain-jafari) [reply src/vivarium_gates_child_iv_iron/components/observers.py:31]: This was an earlier commit in the branch I branched off of, but the runs have had this uncommented for a while now.

File: src/vivarium_gates_child_iv_iron/components/observers.py (hunk 2 above)
Comment (hussain-jafari) [reply src/vivarium_gates_child_iv_iron/components/observers.py]: Will update.

**Pull request merged**